main/src/cgeo/geocaching/CacheListActivity.java

```diff
@@ -1694,15 +1696,17 @@ public static void startActivityFinder(final Activity context, final String user
     }
 
     private void prepareFilterBar() {
-        if (Settings.getCacheType() != CacheType.ALL || adapter.isFiltered()) {
-            final StringBuilder output = new StringBuilder(Settings.getCacheType().getL10n());
-
+        final boolean cacheTypeFilterActive = Settings.getCacheType() != CacheType.ALL;
+        if (cacheTypeFilterActive || adapter.isFiltered()) {
+            final List<String> filters = new ArrayList<>();
+            if (cacheTypeFilterActive) {
```

Comment:
I don't like the duplication of the conditions.

Can you extract the whole if block into an own function that takes the filter name as parameter? You can then call the new function with `Settings.getCacheType().getL10n()` and `adapter.getFilterName()`, respectively.